src/java.base/unix/classes/java/lang/ProcessEnvironment.java

```diff
@@ -99,7 +99,7 @@ static Map<String,String> environment() {
 
     /* Only for use by Runtime.exec(...String[]envp...) */
     static Map<String,String> emptyEnvironment(int capacity) {
-        return new StringEnvironment(new HashMap<>(capacity));
+        return new StringEnvironment(HashMap.newHashMap(capacity));
```

Comment:
This change is correct, since this is called with the length of an array that's used to populate the environment. It really should be named `size` instead of `capacity`. However the windows version of this code simply calls `super(capacity)` as it's a subclass of `HashMap`, which is wrong. Well, probably not worth changing this now. We may need to revisit this later to do some cleaning up. (And also the strange computation in the static initializer at line 71.)

Reply:
@stuart-marks OK, changes on this class reverted.